src/lxp/threat.lua

```diff
@@ -376,7 +376,7 @@ function threat.new(callbacks, separator, merge_character_data)
 					return threat_error("prefix too long")
 				end
 
-				if checks.namespaceUri and #namespaceUri > checks.namespaceUri then
+				if checks.namespaceUri and namespaceUri and #namespaceUri > checks.namespaceUri then
```

Comment:
Can you elucidate a bit on why this change is valid/needed? I am far from an XML expert but it seems counter intuitive to me that we're both specifically being asked to check the namespaceUri (if checks.namespaceUri) and also allowing nil as a valid number that is greater than the number we're supposed to check for. I'm not saying I know this approach is invalid, but I think it should be annotated somehow so it is clear *why* it is this way.

Reply:
XML has a default namespace, which can be set using `<foo xmlns="namespace-uri">` and namespace prefixes which can be set using `<foo xmlns:bar="namespace-uri">`.  Namespace URIs can be any valid URI or empty.  Thus, by using `<foo xmlns="">`, it is possible to reset the default namespace to none. See https://www.w3.org/TR/xml-names/#defaulting